internal/provider/provider_config.go

```diff
@@ -81,6 +80,7 @@ func (c *apiClient) loadAndValidate(ctx context.Context) diag.Diagnostics {
 			TokenSource: oauth2.StaticTokenSource(token),
 		}
 		diags = c.SetupClient(ctx, &creds)
+		return diags
```

Comment:
we are certain we don't want to continue?

Reply:
AccessToken and Credentials are conflicting, so I think we're safe. The linter caught this, which was odd? So maybe I'm missing something, but because they conflict, I think we should be safe. Unless there's a path I'm missing that you can see?

Reply:
The linter would catch something like this, it doesn't know that the two paths can't co-exist in our application. I just wanted to double check 👍 LGTM